### PR TITLE
Add Swift 4.1 compatibility

### PR DESCRIPTION
**Pull request by asmallteapot**

### CODE REVIEW

**Comment by asmallteapot** [comment]
cc @sebastianv1 @zadr

File: Cleanse/CanonicalRepresentable.swift [hunk 1]
```diff
@@ -57,12 +59,3 @@ extension Optional : CanonicalRepresentable {
     }
     
 #endif
-
```
Comment (sebastianv1) [inline Cleanse/CanonicalRepresentable.swift]: This a breaking change to the public API. What was the motivation behind removing it?

Comment (asmallteapot) [reply Cleanse/CanonicalRepresentable.swift]: `ImplicitlyUnwrappedOptional` has been removed as of Swift 4.1. This is necessary for the library to continue to build on that version and newer. Consumers will also have to migrate their code from using `ImplicitlyUnwrappedOptional` when adopting Swift 4.1 or greater.

Comment (sebastianv1) [reply Cleanse/CanonicalRepresentable.swift]: Silly me, this was commented out. Disregard.

**Comment by holmes** [comment]
Looks like we need to migrate the tests as well:

```
Test Case '-[CleanseTests.CanonicalRepresentableTests testCanonicalRepresentable]' started.
Fatal error: 'try!' expression unexpectedly raised an error: 
Multiple Errors
error 1/2:
*** ImplicitlyUnwrappedOptional<NSString> *** binding missing
  -> required by TestRoot at CleanseTests/CanonicalRepresentableTests.swift:39
error 2/2:
*** ImplicitlyUnwrappedOptional<String> *** binding missing
  -> required by TestRoot at CleanseTests/CanonicalRepresentableTests.swift:39
: file /BuildRoot/Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-902.0.48/src/swift/stdlib/public/core/ErrorType.swift, line 184
2018-07-16 19:19:11.052530+0000 xctest[16123:29906] Fatal error: 'try!' expression unexpectedly raised an error: 
Multiple Errors
error 1/2:
*** ImplicitlyUnwrappedOptional<NSString> *** binding missing
  -> required by TestRoot at CleanseTests/CanonicalRepresentableTests.swift:39
error 2/2:
*** ImplicitlyUnwrappedOptional<String> *** binding missing
  -> required by TestRoot at CleanseTests/CanonicalRepresentableTests.swift:39
: file /BuildRoot/Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-902.0.48/src/swift/stdlib/public/core/ErrorType.swift, line 184
```

**Comment by asmallteapot** [comment]
this is superseded by #78

**Pull request closed without merging**